app/scripts/models/account.js

```diff
@@ -323,7 +328,15 @@ define(function (require, exports, module) {
       .then(function (updatedSessionData) {
         self.set(updatedSessionData);
 
-        if (! self.get('verified')) {
+        // `recoveryEmailStatus` does not return a reason why the
+        // sessionToken is not valid, only that it is invalid.
```

Comment:
@vbudhram - now I see, `emailVerified` and `sessionVerified` is different to what I was expecting. I make use of `verificationReason`. I can convert the combination of the two to `verificationReason`, but that may be better to do on the backend.